src/structs/Frames.jl

```diff
@@ -95,3 +96,99 @@ function get_frames(parent, elem, glob::GFrames, last_frames::UnitRange; is_firs
     end
     return glob.frames
 end
+
+"""
+    function get_frames(parent, elem, func_frames::Function, last_frames::UnitRange; is_first = false)
+
+Return the frames based on a specified function. The function `func_frames` is simply evaluated 
+"""
+function get_frames(
+    parent,
+    elem,
+    func_frames::Function,
+    last_frames::UnitRange;
+    is_first = false,
+)
+    return func_frames()
+end
+
+"""
+    prev_start()
+
+The start frame of the previous object or for an action the start frame of the parental object.
+Can be used to provide frame ranges like:
+```
+@Frames(prev_start(), 10)
+```
+"""
+function prev_start()
+    if CURRENT_OBJECT_ACTION_TYPE[1] == :Object
+        PREVIOUS_OBJECT[1].frames.frames[1]
+    else
+        PREVIOUS_ACTION[1].frames.frames[1]
+    end
+end
+
+"""
+    prev_end()
+
+The end frame of the previous object or for an action the end frame of the parental object.
+Can be used to provide frame ranges like:
+```
+@Frames(prev_end()-10, 10)
+```
+"""
+function prev_end()
+    if CURRENT_OBJECT_ACTION_TYPE[1] == :Object
+        PREVIOUS_OBJECT[1].frames.frames[end]
+    else
+        PREVIOUS_ACTION[1].frames.frames[end]
+    end
+end
+
+startof(oa::Union{AbstractAction,AbstractObject}) = oa.frames.frames[1]
+endof(oa::Union{AbstractAction,AbstractObject}) = oa.frames.frames[end]
```

Comment:
I am confused by the usefulness of these shorthands as right now, I am unable to get frames starts and end values from inputted Objects. Where can this be used?

Reply:
You can do:
```julia
obj1 = Object(@Frames(prev_start(), 10), (args...) -> 1)
obj2 = Object(2:100,  (args...) -> 1)
obj3 = Object(@Frames(startof(obj2) + 1, 10), (args...) -> 1)
```        
previously one would have to write `obj3` before `obj2` to be able to use `prev_start()` which most often is probably a good idea anyway to have them in order they appear but in certain cases it might be better to specify the object directly.